code/modules/mob/mob.dm

```diff
@@ -920,3 +922,23 @@
 /mob/setMovetype(newval)
 	. = ..()
 	update_movespeed(FALSE)
+
+/mob/add_trait(id)
+	. = ..()
+	if(.)
+		update_movespeed()
+		if(id == TRAIT_FLIGHT)
+			update_movement_type()
```

Comment:
i keep having to remove those

stop snowflaking the add_trait proc, use wrappers

Reply:
can we just get an on_trait_add proc or nah

Reply:
You're probably better off using an `add_flight` and `remove_flight` proc, check out the fakedeath one if you want an example.
Alternatively if you want to condense it into one make a `set_flight(TRUE/FALSE, source)`.

I'm trying to keep traits as close as possible to read-only "flags", or the snowflake code will pile up rapidly if more systems try to check for a specific trait update to call their own proc.